### PR TITLE
[Merged by Bors] - feat: port Order.Hom.CompleteLattice

### PR DESCRIPTION
**Pull request by LukasMias**
Lot of dangerous instances that were tamed with replacements à la `[CompleteLattice  x] -> {_ : CompleteLattice x}`.  Surprisingly smoothly otherwise, perhaps the translation of `Sup_hom` into `SupₛHom` is controversial.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Hom/CompleteLattice.lean [hunk 1]
```diff
@@ -0,0 +1,1005 @@
...
+
+/-- The map `(a, b) ↦ a ⊓ b` as an `InfₛHom`. -/
+def infInfₛHom : InfₛHom (α × α) α where
+  toFun x := x.1 ⊓ x.2
+  map_infₛ' s := by simp_rw [Prod.fst_infₛ, Prod.snd_infₛ, infₛ_image, infᵢ_inf_eq]
+#align inf_Inf_hom infInfₛHom
+
+@[simp, norm_cast]
+theorem sSupₛHom_apply : supSupₛHom x = x.1 ⊔ x.2 :=
+  rfl
```
Comment (dupuisf) [inline Mathlib/Order/Hom/CompleteLattice.lean:999]:
```suggestion
theorem supSupₛHom_apply : supSupₛHom x = x.1 ⊔ x.2 :=
  rfl
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Order/Hom/CompleteLattice.lean [hunk 2]
```diff
@@ -0,0 +1,1005 @@
...
+/-- The map `(a, b) ↦ a ⊓ b` as an `InfₛHom`. -/
+def infInfₛHom : InfₛHom (α × α) α where
+  toFun x := x.1 ⊓ x.2
+  map_infₛ' s := by simp_rw [Prod.fst_infₛ, Prod.snd_infₛ, infₛ_image, infᵢ_inf_eq]
+#align inf_Inf_hom infInfₛHom
+
+@[simp, norm_cast]
+theorem supSupₛHom_apply : supSupₛHom x = x.1 ⊔ x.2 :=
+  rfl
+#align sup_Sup_hom_apply sSupₛHom_apply
```
Comment (jcommelin) [inline Mathlib/Order/Hom/CompleteLattice.lean:1000]:
```suggestion
#align sup_Sup_hom_apply supSupₛHom_apply
```

**Pull request closed without merging**